### PR TITLE
Added event-analyzer

### PR DESCRIPTION
**Pull request by 5hubh49**
This is how analysis would look like:

Screenshot 2024-08-01 at 9 04 48 PM

### CODE REVIEW

**Comment by 5hubh49** [comment]
Story: [109475](https://app.shortcut.com/replicated/story/109475/add-oomkilled-to-analyzers-in-default-spec)

As per this [doc](https://troubleshoot.sh/docs/analyze/event/), analyzer has been added in the troubleshoot-spec default.yaml

File: in-cluster/default.yaml [hunk 1]
```diff
@@ -468,3 +468,15 @@ spec:
...
+        namespace: default
+        reason: "OOMKilling"
+        kind: Node
+        outcomes:
+          - fail:
+              when: "true"
+              message: Event {{ .Reason }} by object {{ .InvolvedObject.Name }} kind {{ .InvolvedObject.Kind }} has message {{ .Message }}
+          - pass:
+              when: "false"
+              message: No OOMKilled event detected
```
Comment (xavpaice) [inline in-cluster/default.yaml:482]: maybe we should change the wording here to match the event

Comment (5hubh49) [reply in-cluster/default.yaml]: updated

File: in-cluster/default.yaml [hunk 2]
```diff
@@ -468,3 +468,15 @@ spec:
...
               message: "No PVCs are using more than 80% of storage"
+    - event:
+        checkName: event-oom-check
+        namespace: default
+        reason: "OOMKilling"
+        kind: Node
+        outcomes:
+          - fail:
+              when: "true"
+              message: Event {{ .Reason }} by object {{ .InvolvedObject.Name }} kind {{ .InvolvedObject.Kind }} has message {{ .Message }}
```
Comment (banjoh) [inline in-cluster/default.yaml:479]: nit: Perhaps we can have a message like below

```
Pod '{{ .InvolvedObject.Name }}' got killed (OOMKilling) with the reason: {{ .Message }}
```

Comment (xavpaice) [reply in-cluster/default.yaml:479]: given that this event is kind: Node, is the object name not the node name?  I guess we need an example event to confirm.

**Review by adamancini — APPROVED** [review]

File: in-cluster/default.yaml (hunk 2 above)
Comment (banjoh) [reply in-cluster/default.yaml:479]: Good point.

**Review by banjoh — APPROVED** [review]

File: in-cluster/default.yaml (hunk 2 above)
Comment (banjoh) [reply in-cluster/default.yaml:479]: @5hubh49 perhaps a good follow-up would be detecting which pod. Something like below. I tried to get a pod being OOMKilled, and emit an event, but couldn't

```yaml
    - event:
        checkName: OOMkilled detector
        namespace: default
        reason: "OOMKilled"
        kind: Pod
        outcomes:
          - fail:
              when: "true"
              message: Pod '{{ .InvolvedObject.Name }}' got killed with reason: {{ .Message }}
          - pass:
              when: "false" 
              message: No OOM killed pods found
```

**Pull request merged**